server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSPerformSPOptionsCheckUtility.java

```diff
@@ -0,0 +1,282 @@
+                }
+            };
+    public CICSPerformSPOptionsCheckUtility(
+            DialectProcessingContext context, List<SyntaxError> errors) {
+        super(context, errors, DUPLICATE_CHECK_OPTIONS);
+    }
+
+    /**
+     * Entrypoint to check CICS Perform System Command rule options
+     *
+     * @param ctx ParserRuleContext subclass containing options
+     * @param <E> A subclass of ParserRuleContext
+     */
+    public <E extends ParserRuleContext> void checkOptions(E ctx) {
+        switch (ctx.getRuleIndex()) {
+            case RULE_cics_perform_dump:
+                checkDump((CICSParser.Cics_perform_dumpContext) ctx);
+                break;
+            case RULE_cics_perform_endaffinity:
+                checkEndAffinity((CICSParser.Cics_perform_endaffinityContext) ctx);
+                break;
+            case RULE_cics_perform_jvmserver:
+                checkJvmServer((CICSParser.Cics_perform_jvmserverContext) ctx);
+                break;
+            case RULE_cics_perform_pipeline:
+                checkPipeline((CICSParser.Cics_perform_pipelineContext) ctx);
+                break;
+            case RULE_cics_perform_secdiscovery:
+                checkSecdiscovery((CICSParser.Cics_perform_secdiscoveryContext) ctx);
+                break;
+            case RULE_cics_perform_security:
+                checkSecurity((CICSParser.Cics_perform_securityContext) ctx);
+                break;
+            case RULE_cics_perform_shutdown:
+                checkShutdown((CICSParser.Cics_perform_shutdownContext) ctx);
+                break;
+            case RULE_cics_perform_ssl:
+                checkSsl((CICSParser.Cics_perform_sslContext) ctx);
+                break;
+            case RULE_cics_perform_statistics:
+                checkstatistics((CICSParser.Cics_perform_statisticsContext) ctx);
+                break;
+            default:
+                break;
+        }
+        checkDuplicates(ctx);
+    }
+    private void checkDump(CICSParser.Cics_perform_dumpContext ctx) {
+        checkHasMandatoryOptions(ctx.DUMP(), ctx, "DUMP");
+        checkHasMandatoryOptions(ctx.DUMPCODE(), ctx, "DUMPCODE");
+        if (!ctx.TITLE().isEmpty() || !ctx.TITLELENGTH().isEmpty()) {
+            checkHasMandatoryOptions(ctx.TITLE(), ctx, "TITLE");
+            checkHasMandatoryOptions(ctx.TITLELENGTH(), ctx, "TITLELENGTH");
+        }
+        if (!ctx.CALLER().isEmpty() || !ctx.CALLERLENGTH().isEmpty()) {
+            checkHasMandatoryOptions(ctx.CALLER(), ctx, "CALLER");
+            checkHasMandatoryOptions(ctx.CALLERLENGTH(), ctx, "CALLERLENGTH");
+        }
+    }
+    private void checkEndAffinity(CICSParser.Cics_perform_endaffinityContext ctx) {
+        checkHasMandatoryOptions(ctx.ENDAFFINITY(), ctx, "ENDAFFINITY");
+        checkHasMandatoryOptions(ctx.NETNAME(), ctx, "NETNAME");
+    }
+    private void checkJvmServer(CICSParser.Cics_perform_jvmserverContext ctx) {
```

Comment:
I don't think that a diagnostic is produce if I combine e.g. valid DUMP clause with APPIDLEN keyword.